tutorials/27_First_RAG_Pipeline.ipynb

```diff
@@ -0,0 +1,2555 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "2OvkPji9O-qX"
+   },
+   "source": [
+    "# Tutorial: Creating Your First QA Pipeline with Retrieval-Augmentation\n",
+    "\n",
+    "- **Level**: Beginner\n",
+    "- **Time to complete**: 10 minutes\n",
+    "- **Nodes Used**: `InMemoryDocumentStore`, `InMemoryBM25Retriever`, `PromptBuilder`, `GPTGenerator`\n",
+    "- **Prerequisites**: You must have an API key from an active OpenAI account as this tutorial is using the gpt-3.5-turbo model by OpenAI.\n",
+    "- **Goal**: After completing this tutorial, you'll have learned the new prompt syntax and how to use PromptBuilder and GPTGenerator to build a generative question-answering pipeline with retrieval-augmentation.\n",
+    "\n",
+    "> This tutorial uses Haystack 2.0 Beta. To learn more, read the [ Haystack 2.0 Beta announcement](https://haystack.deepset.ai/blog/introducing-haystack-2-beta-and-advent) or see [Haystack 2.0 Documentation](https://docs.haystack.deepset.ai/v2.0/docs)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "LFqHcXYPO-qZ"
+   },
+   "source": [
+    "## Overview\n",
+    "\n",
+    "This tutorial shows you how to create a generative question-answering pipeline using the retrieval-augmentation ([RAG](https://www.deepset.ai/blog/llms-retrieval-augmentation)) approach with Haystack 2.0. The process involves three main components: InMemoryBM25Retriever for fetching relevant documents, PromptBuilder for creating a template prompt, and GPTGenerator for generating responses.\n",
```

Comment:
nit: it might be convenient for the reader to add links to the docs pages for these components

Reply:
agree

Reply:
👍